website/docs/components/file_storage/file_storage_plugin.md

```
@@ -111,3 +112,94 @@ The `removeMany` method receives an array of keys. The plugin maps over the arra
+| `lockFactory` | `ILockFactory`                     | _(required)_                                                                                                                          | A factory that creates named locks                 |
+| `onlyMethods` | `Array<keyof ISignedFileStorageAdapter>` | All methods                                                                                                                      | The subset of methods to protect with a lock       |
+
+:::danger
+Because `withPlugin` uses `enhance` under the hood, the same edge case applies: if one enhanced method internally calls another enhanced method via `this`, the middleware will apply **twice**. Be mindful of inter-method calls when applying plugins that enhance multiple methods on the same instance.
+:::
+
+:::info
+For more information about the `withPlugin` function and applying plugins to adapters, see the [Middleware plugin](/docs/components/middleware#plugin) documentation.
+For more information about lock factories, see the [Lock](../lock/lock_usage.md) documentation.
+:::
+```
```

markdownlint-cli2 (0.23.0)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)